### ISSUE TITLE
ResolvingParser dereferencing local references with RESOLVE_FILES option.

### ISSUE THREAD

**Issue by Megaxela** [jfinkhaeuser/prance#65]
## Expected Behaviour
`ResolvingParser` won't dereference local references when `resolve_types` option is set to `prance.util.resolver.RESOLVE_FILES`. 

### Minimal Example Spec
```yaml
openapi: "3.0.0"
info:
  title: ''
  version: '1.0.0'
paths: {}
components:
    schemas:
        SampleArray:
            type: array
            items:
              $ref: '#/components/schemas/ItemType'

        ItemType:
          type: integer
```

## Actual Behaviour
`ResolvingParser` resolves internal references.

## Steps to Reproduce
Assume that minimal spec file is in `D:/test.yaml`.

```python
import prance
import prance.util.resolver
parser = prance.ResolvingParser(url='D:\\test.yaml', resolve_types=prance.util.resolver.RESOLVE_FILES)
parser.specification
{'openapi': '3.0.0', 'info': {'title': '', 'version': '1.0.0'}, 'paths': {}, 'components': {'schemas': {'SampleArray': {'type': 'array', 'items': {'type': 'integer'}}, 'ItemType': {'type': 'integer'}}}}
```

## Environment
- OS: Any
- Python version: 3.7 / 3.8
- Swagger/OpenAPI version used: OpenAPI
- Backend: openapi-spec-validator

## What I managed to find
I guess, problem is [here](https://github.com/jfinkhaeuser/prance/blob/fdef144b730aa15d7950d526dfb236cf28cb7b22/prance/util/resolver.py#L147). Prance trying to compare `str` and `ParseResult` objects.

@jfinkhaeuser

**Comment by jfinkhaeuser** [comment]
Thanks for this! I'll take a look.

**Comment by jfinkhaeuser** [comment]
Looks like you were right. The tests didn't cover that because they were testing the internal `RefResolver` only, not its usage in the `ResolvingParser`.

**Comment by jfinkhaeuser** [comment]
The fix should be in `v0.18.2` once the build is through.
